### DISCUSSION TITLE
Destructure data from fetch response

### DISCUSSION THREAD

**Discussion by chrissyast** [nuxt/nuxt#19970] [Questions] [answered]
`const { data, pending } = await useLazyAsyncFetch('key', () => $fetch(url))`

the `data` of the api call is like this 
```js
{
  results: [] // the data I care about,
  ...
}
```

How do I get the `results` from the response and set them as a variable that my template can use?

**Comment by wobsoriano** [comment, marked as answer]
Check if `data` is not `null`

```vue
<script setup lang="ts">
  const { data, pending } = await useLazyAsyncFetch('key', () => $fetch(url))
</script>

<template>
  <SomeComponent v-if="data" :results="data.results" />
</template>
```

or maybe make a computed out of it

```vue
<script setup lang="ts">
  const { data, pending } = await useLazyAsyncFetch('key', () => $fetch(url))

  const results = computed(() => data.value?.results ?? [])
</script>

<template>
  <SomeComponent :results="results" />
</template>
```

**Reply by chrissyast** [reply]
Thanks! I thought maybe there was a way to destructure in the same line but this suits fine. Is it essential that typescript is used?

**Reply by wobsoriano** [reply]
I just thought you're using TS

**Reply by chrissyast** [reply]
Sorry, one more question - how would I make this `results` constant editable? If it's a computed property it will be immutable.
